frontend/src/lib/auth/permissions.test.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  STATUSES_ELIGIBLE_FOR_EDIT,
+  STATUSES_ELIGIBLE_FOR_REVIEW,
+} from "../../pages/destructionlist/detail/constants";
+import { User } from "../api/auth";
+import {
+  DESTRUCTION_LIST_STATUSES,
+  DestructionList,
+} from "../api/destructionLists";
+import {
+  canMarkListAsFinal,
+  canReviewDestructionList,
+  canStartDestructionList,
+  canUpdateDestructionList,
+  canViewDestructionList,
+} from "./permissions";
+
+/**
+ * Helper function to generate user objects with different permissions
+ * @param {boolean} canStartDestruction - Permission to start destruction
+ * @param {boolean} canReviewDestruction - Permission to review destruction
+ * @returns {User} - User object with specified permissions
+ */
+const generateUserWithPermissions = (
```

Comment:
Great idea to have factoreis for this, can we to make this even better:

- Move this to `src/fixtures/user.ts`
- Rename this to `userFactory`
- Repurpose this to generic factory:
  - Take an object,
  - Apply it the base (user) fixture (Object.assign)
  - Return  the result
- Use the factory created the correct user:  

```
const permittedUser = userFactory({{
  name: "Test Role",
  canStartDestruction,
  canReviewDestruction,
  canViewCaseDetails: true,
}})
```